src/hooks/useCollections.ts

```diff
@@ -1,55 +1,90 @@
-import { useCallback, useEffect, useState, useMemo } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import { type ApiError, type LoadingState } from '../types';
 import type { CollectionsResponse } from '../types/stac';
 import debounce from '../utils/debounce';
 import { useStacApiContext } from '../context';
 
 type StacCollectionsHook = {
-  collections?: CollectionsResponse,
-  reload: () => void,
-  state: LoadingState
-  error?: ApiError
+  collections?: CollectionsResponse;
+  reload: () => void;
+  state: LoadingState;
+  error?: ApiError;
+  nextPage?: () => void;
+  prevPage?: () => void;
+  setOffset: (newOffset: number) => void;
 };
 
-function useCollections(): StacCollectionsHook {
+export default function useCollections(opts?: {
+  limit?: number;
+  initialOffset?: number;
+}): StacCollectionsHook {
+  const { limit = 10, initialOffset = 0 } = opts || {};
+
   const { stacApi, collections, setCollections } = useStacApiContext();
-  const [ state, setState ] = useState<LoadingState>('IDLE');
-  const [ error, setError ] = useState<ApiError>();
+  const [state, setState] = useState<LoadingState>('IDLE');
+  const [error, setError] = useState<ApiError>();
+
+  const [offset, setOffset] = useState(initialOffset);
+
+  const [hasNext, setHasNext] = useState(false);
+  const [hasPrev, setHasPrev] = useState(false);
 
   const _getCollections = useCallback(
-    () => {
+    async (offset: number, limit: number) => {
       if (stacApi) {
         setState('LOADING');
 
-        stacApi.getCollections()
-          .then(response => response.json())
-          .then(setCollections)
-          .catch((err) => {
-            setError(err);
-            setCollections(undefined);
-          })
-          .finally(() => setState('IDLE'));
+        try {
+          const res = await stacApi.getCollections({ limit, offset });
+          const data: CollectionsResponse = await res.json();
+
+          setHasNext(!!data.links?.find((l) => l.rel === 'next'));
+          setHasPrev(
+            !!data.links?.find((l) => ['prev', 'previous'].includes(l.rel))
+          );
+
+          setCollections(data);
+        } catch (err: any) {
+          setError(err);
+          setCollections(undefined);
+        } finally {
+          setState('IDLE');
+        }
       }
     },
     [setCollections, stacApi]
   );
-  const getCollections = useMemo(() => debounce(_getCollections), [_getCollections]);
 
-  useEffect(
-    () => {
-      if (stacApi && !error && !collections) {
-        getCollections();
-      }
-    },
-    [getCollections, stacApi, collections, error]
+  const getCollections = useCallback(
+    (offset: number, limit: number) =>
+      debounce(() => _getCollections(offset, limit))(),
+    [_getCollections]
   );
 
+  const nextPage = useCallback(() => {
+    setOffset(offset + limit);
```

Comment:
If the collections response has next and/or previous links specified, why are we using `offset` and `limit` here instead of calling the provided links. Also, the [collections spec](https://github.com/radiantearth/stac-api-spec/blob/main/ogcapi-features/README.md#collection-pagination) uses `page` query params for pagination in collections.

Reply:
Hmm, this is unfortunate. I'll check about using the prev/next